pkg/controller/registry/grpc/source.go

```diff
@@ -100,10 +105,70 @@ func (s *SourceStore) Get(key registry.CatalogKey) *SourceConn {
 	return &source
 }
 
+func grpcProxyURL(addr string) (*url.URL, error) {
+	// Handle ip addresses
+	host, _, err := net.SplitHostPort(addr)
+	if err != nil {
+		return nil, err
+	}
+
+	url, err := url.Parse(host)
+	if err != nil {
+		return nil, err
+	}
+
+	// Hardcode fields required for proxy resolution
+	url.Host = addr
+	url.Scheme = "http"
+
+	// Override HTTPS_PROXY and HTTP_PROXY with GRPC_PROXY
+	proxyConfig := &httpproxy.Config{
+		HTTPProxy:  getGRPCProxyEnv(),
+		HTTPSProxy: getGRPCProxyEnv(),
+		NoProxy:    getEnvAny("NO_PROXY", "no_proxy"),
+		CGI:        os.Getenv("REQUEST_METHOD") != "",
+	}
```

Comment:
Is there a way for us to accomplish this _without_ exposing yet-another-knob ™️ to upstream OLM? e.g. downstream-only sidecar container.

Just seems like a lot of networking can be handled at a higher abstraction level (container/pod network).

Reply:
Ideally this would have been accomplished with the `HTTP_PROXY` environment variable, which grpc-go already supports. Unfortunately, that "knob" is already dedicated to our disconnected environment strategy in which all off cluster requests are routed through a proxy. Connecting to the registry pods through that proxy introduces changed behavior on those environments.

Reply:
> Ideally this would have been accomplished with the HTTP_PROXY environment variable, which grpc-go already supports. 

to be clear, I don't think the way we've implemented HTTP proxy support is "at the right level" either.

Reply:
So, my point here is that I don't think the clients here need to know what the proxies are if we have the ability to control networking at other scopes.

Maybe I'm misunderstanding the use case though.

Reply:
There are solutions that exist that could be used to route all traffic from the pod (istio and linkerd) but each of these solutions require the presence of a control plane to facilitate the routing. Short of introducing and supporting a new project that focuses on lightweight network routing, there does not seem to be a tool available that we could use to route GRPC connections.

Reply:
fair enough.